packages/service-library/src/servicelib/background_task.py

```diff
@@ -53,11 +53,17 @@ async def start_periodic_task(
         )
 
 
-async def stop_periodic_task(asyncio_task: asyncio.Task) -> None:
+async def stop_periodic_task(
+    asyncio_task: asyncio.Task, *, timeout: Optional[datetime.timedelta] = None
+) -> None:
     with log_context(
         logger,
         logging.INFO,
         msg=f"cancel periodic background task '{asyncio_task.get_name()}'",
     ), contextlib.suppress(asyncio.CancelledError):
         asyncio_task.cancel()
-        await asyncio_task
+        with log_catch(logger, reraise=False):
+            if timeout:
+                await asyncio.wait_for(asyncio_task, timeout=timeout.total_seconds())
```

Comment:
suggestion:
from python documentation:
"Note that this function does not raise TimeoutError. Futures or Tasks that aren’t done when the timeout occurs are simply returned in the second set."
and passing timeout=None is valid it will just not timeout at all

replace all lines with this simple code snippet
```python
with log_catch(logger, reraise=False):
    await asyncio.wait(asyncio_task, timeout=timeout)
```

Reply:
Thanks for the suggestion. I also made a stupid mistake in not understanding the usage. A sequence of futures is required as input.